bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/util/StringUtil.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smarther.internal.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link StringUtil} class defines common String utility functions, which are used across the whole binding.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@NonNullByDefault
+public class StringUtil {
+
+    private static final int EOF = -1;
+    private static final int DEFAULT_BUFFER_SIZE = 1024 * 4;
+
+    /**
+     * Checks if a String is whitespace, empty ("") or null.
+     *
+     * @param str the String to check, may be null
+     * @return true if the String is null, empty or whitespace
+     */
+    public static boolean isBlank(@Nullable String str) {
+        int strLen;
+        if (str == null || (strLen = str.length()) == 0) {
+            return true;
+        }
+        for (int i = 0; i < strLen; i++) {
+            if (!Character.isWhitespace(str.charAt(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns either the passed in String, or if the String is whitespace, empty ("") or null, the value of defaultStr.
+     *
+     * @param str the String to check, may be null
+     * @param defaultStr the default String to return
+     * @return the passed in String, or the default
+     */
+    public static String defaultIfBlank(String str, String defaultStr) {
+        return StringUtil.isBlank(str) ? defaultStr : str;
+    }
+
+    /**
+     * Strips whitespace from the start and end of a String returning null if the String is empty ("") after the strip.
+     *
+     * @param str the String to be stripped, may be null
+     * @return the stripped String, null if whitespace, empty or null String input
+     */
+    public static @Nullable String stripToNull(@Nullable String str) {
+        if (str == null || str.length() == 0) {
```

Comment:
```suggestion
        if (str == null || str.isEmpty()) {
```

Reply:
Need to return the trimmed string, so just left `if (str == null) { return null; }` and changed the rest to use `trim()` and `isEmpty()`. Thanks